src/main/java/com/blackducksoftware/integration/hub/validator/HubServerConfigValidator.java

```diff
@@ -129,11 +113,23 @@ public ValidationResults assertProxyValid() {
         return results;
     }
 
-    public ValidationResults assertCredentialsValid() {
-        final CredentialsValidator credentialsBuilder = new CredentialsValidator();
-        credentialsBuilder.setUsername(username);
-        credentialsBuilder.setPassword(password);
-        return credentialsBuilder.assertValid();
+    // you can specify either username/password OR apiKey
+    public ValidationResults assertCredentialsOrApiKeyValid() {
+        final ValidationResults validationResults = new ValidationResults();
+
+        if (StringUtils.isBlank(apiKey)) {
+            final CredentialsValidator credentialsBuilder = new CredentialsValidator();
+            credentialsBuilder.setUsername(username);
+            credentialsBuilder.setPassword(password);
+            final ValidationResults credentialsResults = credentialsBuilder.assertValid();
+            validationResults.addAllResults(credentialsResults.getResultMap());
```

Comment:
This line needs to move between line 129 and 130.

Reply:
If it is moved then the if on line 127 would not work correctly

Reply:
Yeah, this comment is outdated. I had another comment below that has since been resolved.